### PR TITLE
[enhancement] querier : Add "query_memory_only" to make loki have option to rely only on memory availability like google monarch.

### PR DESCRIPTION
**Pull request by liguozhong**
**What this PR does / why we need it**:
    Add "query_memory_only" to make loki have option to rely only on memory availability like google monarch.

1: momery + store mode
image
2: momery mode
image
**Which issue(s) this PR fixes**:
Fixes nil
**Special notes for your reviewer**:
image
<Monarch: Google’s Planet-Scale In-Memory
Time Series Database>
 https://www.vldb.org/pvldb/vol13/p3181-adams.pdf
**Why is important**
Last week, due to the 1.5-hour unavailability of the s3 service in our area, the loki cluster operator by me was also completely unavailable for 1.5 hours, which had a very negative impact on my work。
In order to solve this problem, I checked After some information on how other companies solve this problem, I found that google's monarch has an in-memory mode that allows monitoring to be available when storage is unavailable. So I think this mode can help loki operators like me.

ref unavailable loki snapshot：
image

image

image

image

image

If there are similar option, when s3 is unavailable, I can adjust the querier to memory mode to downgrade the service of loki, wait for s3 to become available, and then restore to a fully available loki, so that the failure will not be too serious.
Hope similar failures don't happen to other loki operators like me.

### CODE REVIEW

**Comment by liguozhong** [comment]
@dannykopping  review pr please . 
The availability of loki is my most important job, as operators, we really need this option.

**Comment by liguozhong** [comment]
Today s3 is unavailable again for 4 hours。

image

File: pkg/querier/querier.go [hunk 1]
```diff
@@ -49,6 +50,7 @@ type Config struct {
 	Engine                        logql.EngineOpts `yaml:"engine,omitempty"`
 	MaxConcurrent                 int              `yaml:"max_concurrent"`
 	QueryStoreOnly                bool             `yaml:"query_store_only"`
+	QueryMemoryOnly               bool             `yaml:"query_memory_only"`
```
Comment (sandeepsukhani) [inline pkg/querier/querier.go:53]: I think calling it `query_ingester_only` would be clearer and matches what we are trying to do here.
```suggestion
	QueryIngesterOnly               bool             `yaml:"query_ingester_only"`
```

**Review by sandeepsukhani — APPROVED** [review]
I like the idea. Just a minor suggestion on naming but other than that it LGTM.

**Comment by liguozhong** [comment]
> I like the idea. Just a minor suggestion on naming but other than that it LGTM.

done,thanks

**Comment by liguozhong** [comment]
@owen-d review pr please .
The availability of loki is my most important job, as operators, we really need this option.

**Review by sandeepsukhani — APPROVED** [review]
LGTM

**Pull request merged**